internal/controller/mimiralertmanagerconfig/mimiralertmanagerconfig_controller.go

```diff
@@ -45,6 +46,12 @@ func (r *MimirAlertManagerConfigReconciler) Reconcile(ctx context.Context, req c
 
 	log.FromContext(ctx).Info("Running reconcile on MimirAlertManagerConfig")
 
+	mc, err := r.createMimirClient(ctx, amc)
+	if err != nil {
+		// Status is set only on failure to delete (the status is going to be deleted anyway if it succeeds)
```

Comment:
I don't think this comment is relevant, it's been copied from an invocation below